### PR TITLE
Product filter page

### PR DESCRIPTION
**Pull request by vishal804**
Hello,
created Product Page With The Filter Functionality in This 

### In Component Folder 

src/component/card/ProductCard.js
src/component/filter/Filter.js

### In Context Folder 

src/context/data-context.js

### In Pages Folder 

src/pages/productPage/ProductPage.js

### In Reducer Folder 

src/reducer/dataReducer.js

Please Click on Product Page btn on HomePage to See Product Page
[Link Preview](https://deploy-preview-2--shopping-corner.netlify.app/)

Thank You

### CODE REVIEW

File: src/backend/db/products.js [hunk 1]
```diff
@@ -1,30 +1,92 @@
...
+import { productImage } from "../../assets";
 /**
  * Product Database can be added here.
  * You can add products of your wish with different attributes
  * */
 
 export const products = [
+  // {
+  //   _id: uuid(),
+  //   title: "You Can WIN",
```
Comment (MayankKumar10) [inline src/backend/db/products.js]: suggestion: remove commented code from here

File: src/pages/productPage/ProductPage.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+const ProductPage = () => {
+  const { state, dispatch, filteredData } = useData();
+
+  const getData = async () => {
+    const response = await axios.get("/api/products");
+    dispatch({ type: "SET_PRODUCTS", payload: response.data.products });
+  };
+
+  useEffect(() => {
```
Comment (MayankKumar10) [inline src/pages/productPage/ProductPage.js]: If it's not necessary remove this un-used useEffect.

**Review by MayankKumar10 — APPROVED** [review]
Look into these suggestions else everything looks good to me you can merge it.

File: src/backend/db/products.js [hunk 3]
```diff
@@ -1,30 +1,92 @@
...
+  //   author: "Shiv Khera",
```
Comment (Himanshu-Dhawale) [inline src/backend/db/products.js]: You can remove the unnecessary commented code

**Review by Himanshu-Dhawale — APPROVED** [review]
1) The UI looks good
2) The code is also clean.
Just suggested a small change

File: src/backend/db/products.js (hunk 1 above)
Comment (vishal804) [reply src/backend/db/products.js]: Done the changes

File: src/pages/productPage/ProductPage.js (hunk 2 above)
Comment (vishal804) [reply src/pages/productPage/ProductPage.js]: Done the changes

File: src/backend/db/products.js (hunk 3 above)
Comment (vishal804) [reply src/backend/db/products.js]: Done the changes

**Comment by vishal804** [comment]
> Look into these suggestions else everything looks good to me you can merge it.

Thanks For Review

**Comment by vishal804** [comment]
> 1. The UI looks good
> 2. The code is also clean.
>    Just suggested a small change

Thanks For Review

**Pull request merged**